### PR TITLE
Added authorization for Azure AD OIDC

### PR DESCRIPTION
**Pull request by mortenengel**
Added functionality for Azure OIDC using the passport-azure-ad package

### CODE REVIEW

File: packages/medusa-plugin-auth/src/auth-strategies/azure-oidc/admin.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+export function getAzureAdminAuthRouter(azure: AzureAuthOptions, configModule: ConfigModule): Router {
+	return passportAuthRoutesBuilder({
+		domain: 'admin',
+		configModule,
+		authPath: azure.admin.authPath ?? '/admin/auth/azure',
+		authCallbackPath: azure.admin.authCallbackPath ?? '/admin/auth/azure/cb',
+		successRedirect: azure.admin.successRedirect,
+		strategyName: AZURE_ADMIN_STRATEGY_NAME,
+		passportAuthenticateMiddlewareOptions: {
+			scope: azure.store.scope ?? [],
```
Comment (adrien2p) [inline packages/medusa-plugin-auth/src/auth-strategies/azure-oidc/admin.ts:62]: Todo: .admin.scope

**Review by adrien2p — COMMENTED** [review]
I ll have a deeper look soon but could you also include the doc if possible 👌 great addition, thanks

File: packages/medusa-plugin-auth/src/auth-strategies/azure-oidc/admin.ts (hunk 1 above)
Comment (mortenengel) [reply packages/medusa-plugin-auth/src/auth-strategies/azure-oidc/admin.ts:62]: roger, fixed that one immediately
Will look at docs

Comment (mortenengel) [reply packages/medusa-plugin-auth/src/auth-strategies/azure-oidc/admin.ts]: @adrien2p docs have been added

Comment (adrien2p) [reply packages/medusa-plugin-auth/src/auth-strategies/azure-oidc/admin.ts]: you need to chang the `docs/pages/authentication/_meta.json` and `docs/pages/authentication.mdx`

Comment (mortenengel) [reply packages/medusa-plugin-auth/src/auth-strategies/azure-oidc/admin.ts]: @adrien2p done

Comment (adrien2p) [reply packages/medusa-plugin-auth/src/auth-strategies/azure-oidc/admin.ts]: For some reson, the azure does not appear in the left menu and only appears when being on the authentication page or the azure page. Can you check please

Comment (mortenengel) [reply packages/medusa-plugin-auth/src/auth-strategies/azure-oidc/admin.ts]: Adrien I think that's just Vercel not finding changes to the other files and deciding not to rebuild the pages.
It should be fixed by either a full vercel build or by touching all files - do you want me to add some space to all pages files to force a rebuild?

**Comment by adrien2p** [comment]
I will merge this pr as it is, would you be up to open a separate pr for the unit test. Since it is using passport and the internal utils, you should be able to do the same as the one in the google auth 👍

**Pull request merged**